lib/push.js

```diff
@@ -171,6 +171,40 @@ module.exports = function (log, db, config) {
   }
 
   /**
+   * iOS clients don't yet support all commands types, and due to
+   * platform limitations they have to show some bad fallback UX
+   * if they receive an unsupported message type.  Filter out
+   * devices that we know won't respond well to the given command.
+   *
+   * @param {String} command
+   * The command from the push message payload
+   * @param {Device[]} devices
+   * The list of devices to which to send the push.
+   */
+  function filterSupportedDevices(command, devices) {
+    var minVersion
+    switch (command) {
+    case 'sync:collection_changed':
+      // Everything supports this message, short-circuit.
+      return devices
+    case 'fxaccounts:device_connected':
+    case 'fxaccounts:device_disconnected':
+      minVersion = 9.0
+      break
+    default:
+      minVersion = Infinity
+    }
+    return devices.filter(function(device) {
+      var deviceOS = device.uaOS && device.uaOS.toLowerCase()
+      if (deviceOS !== 'ios') {
+        return true
+      }
+      var deviceVersion = device.uaBrowserVersion ? parseFloat(device.uaBrowserVersion) : 0
+      return deviceVersion >= minVersion
```

Comment:
Would it make sense to add some logging here to see how many iOS devices are being sent or not sent notifications? We could potentially use that data to know when X% of users have migrated to proper version and can remove version check? Or is there another way to get this data?